cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -861,8 +863,10 @@ func ensureNodeMatchesMetadataOrDelete(ctx *controllerContext, csr *capi.Certifi
 	// to the node to exist after a node is preempted. In the case that a GCE
 	// node is preempted and new GCE instance is created with the same name,
 	// explicitly delete all bounds to the old node.
-	if err := deleteAllPodsBoundToNode(ctx, nodeName); err != nil {
-		klog.Warningf("Failed to delete all pods bound to node %q: %v", nodeName, err)
+	if ctx.cleanPreemptibles {
+		if err := deleteAllPodsBoundToNode(ctx, nodeName); err != nil {
+			klog.Warningf("Failed to delete all pods bound to node %q: %v", nodeName, err)
+		}
 	}
 
 	if apierrors.IsNotFound(err) {
```

Comment:
can we also flag guard the change in `shouldDeleteNode` below since it was also introduced in the earlier PR?

```
oldInstanceID := node.ObjectMeta.Annotations[InstanceIDAnnotationKey]

if oldInstanceID != "" && newInstanceID != "" && oldInstanceID != newInstanceID {
		klog.Infof("Detected change in instance ID on node %q - Old Instance ID: %q ; New Instance ID: %q", inst.Name, oldInstanceID, newInstanceID)
		return true, nil
	}
```

Reply:
Changed. I also changed the test file to pass `TestShouldDeleteNode`.